src/app/features/actions/assets/sell-asset/sell-asset.component.ts

```diff
@@ -14,29 +20,55 @@ import { Location } from '@angular/common';
   templateUrl: './sell-asset.component.html',
   styleUrl: './sell-asset.component.scss'
 })
-export class SellAssetComponent {
+export class SellAssetComponent implements OnChanges {
   private _assetIndex: number;
-  public priceControl: FormControl<number>;
+  public mainForm: FormGroup<AssetFormViewModel>;
+
+  @Input() assetType: DealType;
 
   constructor(
     private _activatedRoute: ActivatedRoute,
     private _sessionStore: SessionStoreService,
     private _location: Location
   ) {
     this._assetIndex = +this._activatedRoute.snapshot.params['assetIndex'];
-    this.priceControl = new FormControl<number>(null, [Validators.required, Validators.min(0)]);
+    this._initializeForm();
+  }
+
+  private _initializeForm(): void {
+    this.mainForm = new FormGroup({
+      price: new FormControl(null, [Validators.required, Validators.min(0)])
+    });
+  }
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    if (changes.assetType.currentValue) {
+      this._setupVolumeForm();
+    }
   }
 
   public sell() {
-    if (this.priceControl.invalid) {
+    if (this.mainForm.controls.price.invalid) {
       alert($localize`:@@invalidPrice:Please enter a valid price.`);
       return;
     }
 
-    const cf = confirm($localize`:@@sellAssetConfirm:Are you sure you want to sell asset at $${this.priceControl.value}?`);
+    let priceValue = this.mainForm.controls.price.value;
+    
+    if (this.mainForm.value.volume) {
+      priceValue *= this.mainForm.value.volume;
+    }
+
+    const cf = confirm($localize`:@@sellAssetConfirm:Are you sure you want to sell asset at $${priceValue}?`);
     if (cf) {
-      this._sessionStore.sellAsset(this._assetIndex, this.priceControl.value);
+      this._sessionStore.sellAsset(this._assetIndex, priceValue);
       this._location.historyGo(-5); // move back to the main session page
     }
   }
+
+  private _setupVolumeForm(): void {
+    if (this.assetType !== DEAL_TYPE.STOCKS) return;
+
+    this.mainForm.addControl('volume', new FormControl(null, [Validators.required, Validators.min(0)]));
```

Comment:
should have a max validator for the current volume of the asset.

Reply:
@punkrocker178 notice this not updated

Reply:
This will be implement in next PR, since we will need to retrieve asset information in sell-asset component